### ISSUE TITLE
Random data corruption with SetPosition()

### ISSUE THREAD

**Issue by ShirenY** [grow-graphics/gd#18]
Code below moves Node3D up and down  in a cosin curve. 

```go
package main

import (
	"fmt"

	"grow.graphics/gd"
	"grow.graphics/gd/gdextension"
)

type HelloWorld struct {
	gd.Class[HelloWorld, gd.Node3D] `gd:"HelloWorld"`
}

func (w *HelloWorld) Process(godot gd.Context, delta gd.Float) {

	var pos = w.Super().GetPosition()
	pos.SetY(0 + gd.Cos(gd.Time(godot).GetUnixTimeFromSystem()))
	w.Super().SetPosition(pos)

	fmt.Printf("hello %f %f %f\n", pos.X(), pos.Y(), pos.Z())
}

func main() {
	godot, ok := gdextension.Link()
	if !ok {
		panic("could not link to godot")
	}

	gd.Register[HelloWorld](godot)
}
```

But some times the position value goes wild like this.

```
hello 0.000000 -0.349178 0.000000
hello 0.000000 -0.334141 0.000000
hello 0.000000 -0.318071 0.000000
hello 0.000000 -0.316174 0.000000
hello 0.000000 -0.301908 0.000000
hello 0.000000 -0.286617 -6952734638705756535259136.000000
hello 0.000000 -0.270289 -6952734638705756535259136.000000
hello 0.000000 -0.254851 -6952734638705756535259136.000000
hello 0.000000 -0.238376 -6952734638705756535259136.000000
hello 0.000000 -0.220857 0.000000
hello 0.000000 -0.206204 0.000000
hello 0.000000 -0.189540 0.000000
hello 0.000000 -0.154077 0.000000
hello 0.000000 -0.073627 0.000000
hello 0.000000 -0.070635 0.000000
hello 0.000000 -0.056663 0.000000
hello 0.000000 -0.040682 0.000000
hello 0.000000 -0.023691 -6952734638705756535259136.000000
hello 0.000000 -0.007694 -6952734638705756535259136.000000
hello 0.000000 0.010306 -6952734638705756535259136.000000
hello 0.000000 0.026303 0.000000
hello 0.000000 0.060270 0.000000
hello 0.000000 0.077229 0.000000
hello 0.000000 0.092175 0.000000
hello 0.000000 0.110083 0.000000
```

The z component appears to be corrupted. 
In some sessions this behaviour never happens, but others it just goes wild. In the view port it makes the cube attached to it disappears.
I'm I doing something wrong? 
Let me know if you need a minimal reproduce project. Frankly I'm not sure if this can be reproduce on your machine.

**Comment by Splizard** [comment]
Thanks for reporting, I will investigate this, it does look concerning, it may have something to do with the alignment/layout of Vector3.

**Comment by Splizard** [comment]
May be fixed with the latest commit, `go get -u` to update your dependencies and let me know how you get on.

**Comment by ShirenY** [comment]
If I understand it correctly, the new added code equals to 
```
clear(frame.buf[:0])
```
Which is doing nothing. Because frame.len is set to 0 right before it.

**Comment by Splizard** [comment]
Thanks for the catch, I was trying to fix it this morning before work and I didn't notice this, will fix right now.

**Comment by ShirenY** [comment]
I tried the latest code, but unfortunately, there's still data corruption. This time, I changed the node position in the scene to (1, 0, 1) to ensure the data isn't cleared out. When I repeatedly play and stop, the log still shows incorrect data in some sessions.

[log.txt](https://github.com/grow-graphics/gd/files/15307759/log.txt)

**Comment by Splizard** [comment]
Ok, unfortunately I'm not able to reproduce yet, it's odd that in the first log, the value was cleared for a limited number of process executions, before returning to `1`.

Is the behaviour essentially the same as before then? If something has changed then it may still have something to do with the callframe. Otherwise it may be worth running a time travelling debugger to catch a corrupted run and then having a look through this. Are you familiar with debuggers by any chance?

The behaviour seems a bit flaky so it may not be even be possible to capture with a `go test`, it may also be related to the compiler and platform you are on. In any case, it sure looks like some sort of mis-use of unsafe issue. Does it still occur without the sin call?

**Comment by ShirenY** [comment]
This time it feels a bit different, but since this issue is quite random, I'm not sure is it really changed.
I will try to debug it. While I had never debuged go yet, so this might take some time.

**Comment by ShirenY** [comment]
Well, I tried delve and gdb but looks like both of them are not designed to debug a dynamic library. 
https://intellij-support.jetbrains.com/hc/en-us/community/posts/4413795710866-Debug-DLL
Do you know how I can I attach the debuger to godot with go?

**Comment by Splizard** [comment]
I've used gdb successfully before, you debug godot itself (launch godot in the graphics subdirectory inside the debugger) and since gdb supports Go, it typically just works.

If you can setup a recording with gdb, you may even be able to share this with me so I can explore what's happening (in theory, although I'm not 100% sure this works cross platform).

**Comment by ShirenY** [comment]
I will try gdb again by launching the godot directly once I got time. However I'm afraid there maybe some flaws on windows for this process. I will let you know.

**Comment by ShirenY** [comment]
Also this the code to quickly view the result of the size result,
https://go.dev/play/p/QoGBOhh6ilu

**Comment by Splizard** [comment]
Oh wow, that'll be it. Thank you very much for investigating this issue, I really appreciate it.

**Comment by Splizard** [comment]
Have addressed this in 302a276

**Comment by ShirenY** [comment]
Can confirm latest version fixed this issue. Now the cube moves perfectly, thanks!

Is this [change ]( https://github.com/grow-graphics/gd/commit/1ba298be911e7f0c20e54af8681946c5c4ab0255 ) still relevant? Maybe we can revert it if it does not help with this issue, clear can cost some time I guess.
